src/test.ts

```diff
@@ -23,38 +25,57 @@ export const setupTests = async <Paths extends {}>(bindings: Env) => {
       ...bindings,
     },
     queueConsumers: (config.queues?.consumers ?? []).map(
-      ({ queue }: { queue: string }) => queue,
+      ({ queue }: { queue: string }) => queue
     ),
     queueProducers: Object.fromEntries(
       (config.queues?.producers ?? []).map(
         ({ binding, queue }: { queue: string; binding: string }) => [
           binding,
           queue,
-        ],
-      ),
+        ]
+      )
     ),
     r2Buckets: (config.r2_buckets ?? []).map(
-      ({ binding }: { binding: string }) => binding,
+      ({ binding }: { binding: string }) => binding
     ),
     kvNamespaces: (config.kv_namespaces ?? []).map(
-      ({ binding }: { binding: string }) => binding,
+      ({ binding }: { binding: string }) => binding
     ),
     d1Databases: (config.d1_databases ?? []).map(
-      ({ binding }: { binding: string }) => binding,
+      ({ binding }: { binding: string }) => binding
     ),
     serviceBindings: Object.fromEntries(
       (config.services ?? []).map(
         ({ binding, service }: { binding: string; service: string }) => [
           binding,
           service,
-        ],
-      ),
+        ]
+      )
     ),
     fetchMock,
   });
 
   const url = await miniflare.ready;
 
+  // Hack until miniflare supports auto running D1 migrations.
+  // Note: Each migration file can contain only a single statement.
+  await Promise.all(
+    (config.d1_databases ?? []).map(
+      async ({ binding }: { binding: string }) => {
+        const d1 = await miniflare.getD1Database(binding);
+
+        const migrationsDir = "./migrations";
+        for (const migrationName of await nodefs.readdir(migrationsDir)) {
+          const migrationPath = path.join(migrationsDir, migrationName);
+          let migration = await nodefs.readFile(migrationPath, "utf8");
+          // Remove migration comment
+          migration = migration.split("\n").slice(1).join(" ");
```

Comment:
Feels like we are going to trip over more different syntaxes here, we may need to use wrangler to give us more parity; we can try and see.